### PR TITLE
Adds a new traitor objective: bankrupt

### PR DESCRIPTION
**Pull request by ike709**
Adds a new traitor objective: bankrupt. The traitor must ensure the target department (currently only cargo, but admins can edit it) ends the round with less than 1000 credits in their department account.

## Changelog
:cl:
add: New traitor objective: bankrupt. Ensure the target department ends the round with less than 1000 credits in its department account.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -689,6 +689,31 @@ GLOBAL_LIST_EMPTY(possible_items_special)
...
+
+/datum/objective/bankrupt/check_completion()
+	var/datum/bank_account/D = SSeconomy.get_dep_account(target_department)
+	if(isnull(D) || D.account_balance <= target_amount)
+		return TRUE
+
+	return FALSE
+
+/datum/objective/bankrupt/admin_edit(mob/admin)
+	var/dep_choice = input(admin, "Which target department?", "Department Selection") as null|anything in SSeconomy.department_accounts
```
Comment (ike709) [inline code/game/gamemodes/objective.dm:709]: I need to double-check this when it isn't 2am because there's a good chance it won't work.

Comment (ike709) [reply code/game/gamemodes/objective.dm:709]: (I'm actually fairly certain it doesn't)

Comment (ike709) [reply code/game/gamemodes/objective.dm:709]: Actually it probably does.

Comment (ike709) [reply code/game/gamemodes/objective.dm:709]: (This is why you don't code at 2-3am).

**Comment by eeSPee** [comment]
Hmm... Might not work well with non-cargo budgets.

Step 1: ask RD for the cargo budget.
Step 2: give all budget to cargo and be declared a hero.

Also power gaming heads will hog the budgets and validhunt anyone that asks for the budget. 

Cargo might work good though.

File: code/modules/antagonists/traitor/datum_traitor.dm [hunk 2]
```diff
@@ -163,6 +163,12 @@
 			download_objective.owner = owner
 			download_objective.gen_amount_goal()
 			add_objective(download_objective)
+
+		else if(prob(15) && !(locate(/datum/objective/bankrupt) in objectives) && !(owner.assigned_role in list("Captain", "Head of Personnel", "Quartermaster")))
```
Comment (Archanial) [inline code/modules/antagonists/traitor/datum_traitor.dm:167]: Probably every head should be excluded from this

**Comment by FriendlyContractor** [comment]
I dont think this objective will add anything new to the game.

 The only reliable way of completing it will be stealing budget card and taking money out of it right before shuttle docks(also as far as i know budgets generate money over time so you might get screwed over if money is added at bad time) at centcom which means "bankrupt" turns into yet another steal objective where you have to emag a locker(super boring) or murder head in possesion of it

**Comment by Sinestia** [comment]
> I dont think this objective will add anything new to the game.
> 
> The only reliable way of completing it will be stealing budget card and taking money out of it right before shuttle docks(also as far as i know budgets generate money over time so you might get screwed over if money is added at bad time) at centcom which means "bankrupt" turns into yet another steal objective where you have to emag a locker(super boring) or murder head in possesion of it



Yeah, this makes me think it needs work. 

 Given its about stealing money imo there should be some sort of reward and means of getting rid of it, sort of like a contractor-styled objective

**Comment by eeSPee** [comment]
> I dont think this objective will add anything new to the game.
> 
> The only reliable way of completing it will be stealing budget card and taking money out of it right before shuttle docks(also as far as i know budgets generate money over time so you might get screwed over if money is added at bad time) at centcom which means "bankrupt" turns into yet another steal objective where you have to emag a locker(super boring) or murder head in possesion of it

Really good point.

I would rather have it as "Get away with x000 credits."

**Comment by nikk735** [comment]
steal blank credits from whatever department or just credits overall would be cool

edit: maybe assistant only steal blank amount of credits

**Comment by eeSPee** [comment]
No. But it could be a easier goal.

Would be nice on BBs.

File: code/modules/antagonists/traitor/datum_traitor.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/antagonists/traitor/datum_traitor.dm:167]: Would be pretty funny to have heads trying to bankrupt other departments

**Comment by ItsTisme** [comment]
It has been said already, but other then stealing the budget card(crab 19 MABY) there is no way to complete this which makes it bassicly a steal objective. Only being able to steal cargo budget would be fun since it gives purpose to the vault but even that seems annoying. Or some one cheeses and orders a bunch of useless crates.

**Comment by ike709** [comment]
Busy with other things, someone else can take this over if they want.

**Pull request closed without merging**